src/view/com/util/List.web.tsx

```diff
@@ -0,0 +1,281 @@
+    renderItem,
+    extraData,
+    style,
+    ...props
+  }: ListProps<ItemT>,
+  ref: React.Ref<ListMethods>,
+) {
+  const pal = usePalette('default')
+  const {isMobile} = useWebMediaQueries()
+  if (!isMobile) {
+    contentContainerStyle = addStyle(
+      contentContainerStyle,
+      styles.containerScroll,
+    )
+  }
+
+  let header: ListProps<ItemT>['ListHeaderComponent'] = null
+  if (ListHeaderComponent != null) {
+    if (typeof ListHeaderComponent === 'object') {
+      header = ListHeaderComponent
+    } else if (typeof ListHeaderComponent === 'function') {
+      // @ts-ignore We aren't using classes so it's a render function.
+      header = ListHeaderComponent()
+    }
+  }
+
+  let footer: ListProps<ItemT>['ListHeaderComponent'] = null
+  if (ListFooterComponent != null) {
+    if (typeof ListFooterComponent === 'object') {
+      footer = ListFooterComponent
+    } else if (typeof ListFooterComponent === 'function') {
+      // @ts-ignore We aren't using classes so it's a render function.
+      footer = ListFooterComponent()
+    }
+  }
+
+  if (headerOffset != null) {
+    style = addStyle(style, {
+      paddingTop: headerOffset,
+    })
+  }
+
+  const nativeRef = React.useRef(null)
+  React.useImperativeHandle(
+    ref,
+    () =>
+      ({
+        scrollToTop() {
+          window.scrollTo({top: 0})
+        },
+        scrollToOffset({
+          animated,
+          offset,
+        }: {
+          animated: boolean
+          offset: number
+        }) {
+          window.scrollTo({
+            left: 0,
+            top: offset,
+            behavior: animated ? 'smooth' : 'instant',
+          })
+        },
+      } as any), // TODO: Types.
+    [],
+  )
+
+  const [isVisible, setIsVisible] = React.useState(false)
+  React.useEffect(() => {
+    if (!isVisible) {
+      return
+    }
+    function handleScroll() {
+      console.log(window.scrollY)
+    }
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [isVisible])
+
+  const isScrolledDown = useRef(false)
+  function handleAboveTheFoldVisibleChange(isAboveTheFold: boolean) {
+    const didScrollDown = !isAboveTheFold
+    if (isScrolledDown.current !== didScrollDown) {
+      isScrolledDown.current = didScrollDown
+      startTransition(() => {
+        onScrolledDownChange?.(didScrollDown)
+      })
+    }
+  }
+
+  return (
+    <View {...props} style={style} ref={nativeRef}>
+      <Visibility
+        onVisibleChange={setIsVisible}
+        style={styles.parentTreeVisibilityDetector}
+      />
+      <View
+        style={[
+          styles.contentContainer,
+          contentContainerStyle,
+          desktopFixedHeight ? styles.minHeightViewport : null,
+          pal.border,
+        ]}>
+        <Visibility
+          onVisibleChange={handleAboveTheFoldVisibleChange}
+          style={[styles.aboveTheFoldDetector, {height: headerOffset}]}
```

Comment:
This is where I'm using https://github.com/bluesky-social/social-app/pull/2216. I think it's helpful to know at this level that this is a specific concept (`headerOffset`) and not just some padding.